core/src/main/scala/kafka/log/Log.scala

```diff
@@ -918,6 +954,7 @@ class Log(@volatile var dir: File,
                                 preallocate = config.preallocate))
       updateLogEndOffset(newOffset)
       this.recoveryPoint = math.min(newOffset, this.recoveryPoint)
+      this.logStartOffset = math.min(newOffset, this.logStartOffset)
```

Comment:
Thought about this more. I think we should just set `logStartOffset` to `newOffset`. The reason is that `truncateFullyAndStartAt(...)` is supposed to create a new log. Thus `logStartOffset` should be the same as `baseOffset` of the first segment.